install.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env sh
+
+# TODO use ghcup to install ghc/stack
+install_ghcup() {
+  export BOOTSTRAP_HASKELL_NONINTERACTIVE=1
+  export BOOTSTRAP_HASKELL_MINIMAL=1
+  curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
+  . ~/.ghcup/env
+}
+
+install_ghc() {
+  ghcup install ghc 8.10.7 --set
+}
+
+install_cabal() {
+  ghcup install cabal 3.6.2.0 --set
+}
+
+install_stack() {
+  ghcup install stack 2.7.3 --set
+}
+
+install_ghcup
+ghcup upgrade
+install_ghc
+install_cabal
+cabal update
```

Comment:
Do we need to do this in the base image? For one thing, we're not also doing `stack update`, so the Hackage index will only be populated for Cabal. For another, this will just give us the Hackage index at the time that the image was built.

Reply:
What didn't work? 

I would prefer having no Hackage index to having a stale index. With a stale index it's easy to do something like `cabal install hspec-discover` and accidentally get an old version. With no index you'll just get an error.

Reply:
You get an error with:

```
ghcup compile hls -g master --ghc 8.10.7 --set -- --ghc-options='-dynamic'
```

The commit adding haskell-language-server back isn't in this PR yet because I haven't pushed it because I haven't tested it (uggh slow iteration times). 

If you're curious, I'm adding haskell-language-server 1.6 (old version) back so that this PR isn't breaking after I noticed during pairing that @luis-aparicio is still using it and I don't want to break his workflow.

Reply:
I'm confused, my current understanding is:

- If we care about the index, we can set it as early as possible in the base image
- If we don't care about the index, we can use `cabal update` whenever

> With a stale index it's easy to do something like cabal install hspec-discover and accidentally get an old version. With no index you'll just get an error.

Aren't we depending on things like `hspec-discover` to have already been installed?

Perhaps you mean for the use case of someone doing something like `docker compose exec stack cabal install hspec-discover` to try a newer version but silently getting an older one?

Reply:
Oh dear, that could be unfortunate. If we have to compile HLS from source using Cabal, then we're going to have a bunch of extra stuff in this image that we don't use for anything else. I suppose we could blow away `~/.cabal` though.

Reply:
I'll double check we still get an error and have problems not compiling from source.

Reply:
I have it building now @tfausak to test that, but would you prefer:

- avoiding compiling HLS from source in `docker-stack`
- Just `rm -r ~/.cabal` at the end
- something else?

Now that I've tested things end to end are working, I want to figure out the fastest path to getting this merged. Since you're the sole reviewer I need to understand where your head is at so I can avoid too much work re-reviewing for you and addressing PR comments for me.

Reply:
Yep, just verified that without compiling we get spammed with:

```
Message: 
  Internal error, getIdeGlobalExtras, no entry for DisplayTHWarning
  CallStack (from HasCallStack):
  errorIO, called at src/Development/IDE/Core/Shake.hs:351:20 in
  ghcide-1.7.0.0-5d0280f91cc32129977b489cf9b14cb95b9062238f5e7b9a5bc31c595dd97f64:Development.IDE.Core.Shake
```

This translates to the template haskell warning being displayed if called from vscode.

I'll go the approach of nuking `~/.cabal` given this and also add a `stack update` for now.

Feel free to suggest modifications to that, but it's my best guess of what you think is ideal.